File: samples-java/src/main/java/io/github/ilaborie/loops/samples/anatomy/Filter.java
package io.github.ilaborie.loops.samples.anatomy;

import io.github.ilaborie.loops.samples.Element;

import java.util.ArrayList;
import java.util.List;

import static io.github.ilaborie.loops.samples.Elements.getSomeElements;

public class Filter {

    public static void main(String[] args) {
        List<Element> elements = getSomeElements();
        List<Element> result = filter(elements);
        result.forEach(System.out::println);
    }

    public static List<Element> filter(List<Element> elements) {
        List<Element> result = new ArrayList<>();

        for (Element elt : elements) {
            if (elt.getName().startsWith("a")) {
                result.add(elt);
            }
        }

        return result;
    }

}

File: samples-kotlin/src/main/kotlin/io/github/ilaborie/loops/samples/anatomy/filter.kt
package io.github.ilaborie.loops.samples.anatomy

import io.github.ilaborie.loops.samples.ElementKt
import io.github.ilaborie.loops.samples.getSomeElementsKt


fun main() {
    val elements = getSomeElementsKt()
    val result = io.github.ilaborie.loops.samples.stream.filter(elements)
    result.forEach { println(it) }
}

fun filter(elements: List<ElementKt>): List<ElementKt> {
    val result = mutableListOf<ElementKt>()

    for (elt in elements) {
        if (elt.name.startsWith("a")) {
            result.add(elt)
        }
    }

    return result
}